base/applications/rapps/loaddlg.cpp

```diff
@@ -80,6 +81,30 @@ LoadStatusString(DownloadStatus StatusParam)
     return szString;
 }
 
+#define FILENAME_VALID_CHAR ( \
+    PATH_CHAR_CLASS_LETTER      | \
+    PATH_CHAR_CLASS_DOT         | \
+    PATH_CHAR_CLASS_SEMICOLON   | \
+    PATH_CHAR_CLASS_COMMA       | \
+    PATH_CHAR_CLASS_SPACE       | \
+    PATH_CHAR_CLASS_OTHER_VALID)
+
+VOID
+UrlUnescapeAndMakeFileNameValid(CStringW& str)
+{
+    WCHAR szPath[MAX_PATH];
+    DWORD cchPath = _countof(szPath);
+    UrlUnescapeW(const_cast<LPWSTR>((LPCWSTR)str), szPath, &cchPath, 0);
+
+    for (PWCHAR pch = szPath; *pch; ++pch)
+    {
+        if (PathIsValidCharW(*pch, FILENAME_VALID_CHAR))
```

Comment:
Shouldn't this be
```suggestion
        if (!PathIsValidCharW(*pch, FILENAME_VALID_CHAR))
```
if this loop is meant to imitate what's done here:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shell32/wine/shellpath.c;hb=dc4992ce4addc594722b5d23e1c137429b995b7d#l224
??

Reply:
Thanks.